src/gui/editors/AutomationEditor.cpp

```diff
@@ -497,162 +497,159 @@ void AutomationEditor::mousePressEvent( QMouseEvent* mouseEvent )
 	{
 		return;
 	}
-	if( mouseEvent->y() > TOP_MARGIN )
+
+	// If we clicked inside the AutomationEditor viewport (where the nodes are represented)
+	if( mouseEvent->y() > TOP_MARGIN && mouseEvent->x() >= VALUES_WIDTH )
 	{
 		float level = getLevel( mouseEvent->y() );
 
-		int x = mouseEvent->x();
+		// Get the viewport X
+		int x = mouseEvent->x() - VALUES_WIDTH;
 
-		if( x >= VALUES_WIDTH )
-		{
-			// set or move value
+		// Get tick in which the user clicked
+		int posTicks = (x * MidiTime::ticksPerBar()/m_ppb) + m_currentPosition;
 
-			x -= VALUES_WIDTH;
+		// Get the time map of current pattern
+		timeMap & tm = m_pattern->getTimeMap();
 
-			// get tick in which the user clicked
-			int pos_ticks = x * MidiTime::ticksPerBar() / m_ppb +
-							m_currentPosition;
+		// Check if we are clicking over a node (it will be tm.end() if not)
+		timeMap::iterator clickedNode = getNodeAt(mouseEvent->x(), mouseEvent->y());
 
-			// get time map of current pattern
-			timeMap & time_map = m_pattern->getTimeMap();
+		if (mouseEvent->button() == Qt::LeftButton)
+		{
+			m_mouseDownLeft = true;
+		}
+		if( mouseEvent->button() == Qt::RightButton )
+		{
+			m_mouseDownRight = true;
+		}
 
-			// will be our iterator in the following loop
-			timeMap::iterator it = time_map.begin();
+		if( mouseEvent->button() == Qt::LeftButton &&
+						m_editMode == DRAW )
```

Comment:
We just set the variables above, so why not use them? :)
```suggestion
		if (m_mouseDownLeft && m_editMode == DRAW)
```